apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -381,6 +382,14 @@ func (r *OpenStackControlPlane) DefaultServices() {
 
 	// Neutron
 	r.Spec.Neutron.Template.Default()
+	// WIP: Error when trying to trigger webhook by modifying the CR.
+	if r.Spec.Neutron.APIOverride.Route == nil {
```

Comment:
This logic may be helpful for other services, consider introducing a helper function that would receive an override object and default annotations and that would initialize the override route accordingly.

Reply:
This should be defined in lib-common, right? Should I propose it there and then change it here on a follow-up patch? Or is there a better way?

Reply:
You could have it here for now since the code that will use it will be here too, no? If it implies lib-common, definitely don't go there not to slow down merging it. 

(Other projects should probably take it from there and iterate over all service CRs to initialize defaults.)

Reply:
you are right, at first I thought of creating the function with a receiver, and that can only be done from the lib-common repository, but I can pass the * OverrideSpec as an argument.